extensions/variants.go

```diff
@@ -72,33 +72,38 @@ func validateType(arg Argument, actual types.Type, idx int, nullHandling Nullabi
 	return allNonNull, nil
 }
 
-func EvaluateTypeExpression(nullHandling NullabilityHandling, expr types.FuncDefArgType, paramTypeList ArgumentList, variadic *VariadicBehavior, actualTypes []types.Type) (types.Type, error) {
-	if len(paramTypeList) != len(actualTypes) {
+// EvaluateTypeExpression evaluates the function return type given the input argumentTypes
+//
+//	funcParameters: the function parameters as defined in the function signature in the extension
+//	argumentTypes: the actual argument types provided to the function
+func EvaluateTypeExpression(nullHandling NullabilityHandling, returnTypeExpr types.FuncDefArgType,
+	funcParameters FuncParameterList, variadic *VariadicBehavior, argumentTypes []types.Type) (types.Type, error) {
+	if len(funcParameters) != len(argumentTypes) {
 		if variadic == nil {
 			return nil, fmt.Errorf("%w: mismatch in number of arguments provided. got %d, expected %d",
-				substraitgo.ErrInvalidExpr, len(actualTypes), len(paramTypeList))
+				substraitgo.ErrInvalidExpr, len(argumentTypes), len(funcParameters))
 		}
 
-		if !variadic.IsValidArgumentCount(len(actualTypes) - len(paramTypeList) - 1) {
+		if !variadic.IsValidArgumentCount(len(argumentTypes) - len(funcParameters) - 1) {
```

Comment:
Is this code right? It looks wrong.

Reply:
forgot to mention this is fixed added some testcases also.